site/docs/develop/index.md

```diff
@@ -15,7 +15,7 @@ creators.
 The Iceberg format relies on a set of metadata files stored with (or near) the actual
 data tables. This allows Iceberg to fulfill the same role as the Hive Metastore for transactions without the need for
 expensive metadata scans or centralized planning (see [Iceberg
-performance](https://iceberg.apache.org/performance/)). This includes
+performance](https://iceberg.apache.org/docs/nightly/performance/)). This includes
```

Comment:
I'm on the fence with nightly URLs, how about `https://iceberg.apache.org/docs/latest/performance/`?